src/BTVNanoCommissioning/workflows/ctag_DY_valid_sf.py

```diff
@@ -0,0 +1,276 @@
+import gzip
+import pickle, os, sys, mplhep as hep, numpy as np
+import collections
+
+from matplotlib.pyplot import jet
+
+import coffea
+from coffea import hist, processor
+import awkward as ak
+from coffea.analysis_tools import Weights
+
+from BTVNanoCommissioning.helpers.definitions import definitions
+from BTVNanoCommissioning.utils.correction import lumiMasks,muSFs,load_pu,load_BTV
+from BTVNanoCommissioning.utils.AK4_parameters import correction_config
+from BTVNanoCommissioning.helpers.cTagSFReader import getSF
+
+
+
+class NanoProcessor(processor.ProcessorABC):
+    # Define histograms
+    
+
+    def num(ar):
+        return ak.num(ak.fill_none(ar[~ak.is_none(ar)], 0), axis=0)
+    def __init__(self,year="2017",campaign="Rereco17_94X"):        
+        self._year = year
+        self._campaign= campaign
+        # Define axes
+        # Should read axes from NanoAOD config
+        dataset_axis = hist.Cat("dataset", "Primary dataset")
+        flav_axis = hist.Bin("flav", r"Genflavour",[0,1,4,5,6])
+
+        # Jet
+        jet_pt_axis   = hist.Bin("pt",   r"Jet $p_{T}$ [GeV]", 50, 0, 500)
+        jet_eta_axis  = hist.Bin("eta",  r"Jet $\eta$", 25, -2.5, 2.5)
+        jet_phi_axis  = hist.Bin("phi",  r"Jet $\phi$", 30, -3, 3)
+        jet_mass_axis = hist.Bin("mass", r"Jet $m$ [GeV]", 50, 0, 500)
+
+        # Events
+        njet_axis  = hist.Bin("njet",  r"N jets",      [0,1,2,3])
+        ## Muons
+        hard_lpt_axis = hist.Bin("pt",   r"Hard lepton pt", 40, 0, 200)
+        hard_leta_axis  = hist.Bin("eta",  r"Lepton $\eta$", 25, -2.5, 2.5)
+        hard_lphi_axis  = hist.Bin("phi",  r"Lepton $\phi$", 30, -3, 3)
+        hard_liso_axis = hist.Bin("pfRelIso04_all", r"Hard Muon Rel. Iso", 40,0,4.)
+        soft_lpt_axis   = hist.Bin("pt",   r"Soft lepton pt", 40, 0, 40)
+        soft_leta_axis  = hist.Bin("eta",  r"Lepton $\eta$", 25, -2.5, 2.5)
+        soft_lphi_axis  = hist.Bin("phi",  r"Lepton $\phi$", 30, -3, 3)
+        soft_liso_axis = hist.Bin("pfRelIso04_all", r"Soft Muon Rel. Iso", 40,0,4.)
+        l_dxy_axis = hist.Bin("dxy", r"dxy", 20,0,0.002)    
+        l_dz_axis = hist.Bin("dz", r"dz", 20,0,0.01)    
+        l_sip3d_axis = hist.Bin("dz", r"dz", 20,0,0.2)
+        
+
+        ## Z
+        zmass_axis = hist.Bin("zmass", r"Z Mass", 25, 50,100)
+        zpt_axis = hist.Bin("zpt",r"Z $p_{T}$", 25,0,100)
+        zeta_axis  = hist.Bin("zeta",  r"Z $\eta$", 25, -2.5, 2.5)
+        zphi_axis  = hist.Bin("zphi",  r"Z $\phi$", 30, -3, 3)
+        drmumu_axis = hist.Bin("drmumu", r"$\Delta$R($\mu_{soft}$,$\mu_{hard}$)", 25,0,5)
+
+        syst_axis = hist.Cat("syst",['noSF','SF','SFup','SFdn'])
+
+        btagDeeplist = [
```

Comment:
is it the same list as in` definitions.py` ?

Reply:
Do you mean the `btagDeeplist` ? They are the same list

Reply:
Right, could just import it from there then?